### ISSUE TITLE
Analysis failure error is unhelpful

### ISSUE THREAD

**Issue by stewbasic** [bazelbuild/bazel#3646]
I am seeing the following error:

```
ERROR: Analysis of target '//android:android' failed; build aborted
INFO: Elapsed time: 0.629s
FAILED: Build did NOT complete successfully (2 packages loaded)
```

Presumably a skylark rule or macro that I depend on threw an error, but bazel doesn't tell me which or where. Such information would aid further debugging.

My specific problem seems to be caused by

```
android_ndk_repository(
    name = "androidndk",
    api_level = 16,
)
```

but regardless of what is wrong in my case, it would be generally useful to have more information provided when the analysis phase fails.

Note: I didn't find any analysis verbosity option in `bazel help build` (--explain doesn't produce anything). Please close the issue if there is an option that I missed. 

### Environment info
* Operating System: ubuntu 16.04 LTS
* Bazel version (output of `bazel info release`): release 0.5.4

**Comment by aj-michael** [comment]
@jin Perhaps this is related to #3740?

**Comment by jin** [comment]
Seems like it.

@stewbasic, did you end up finding what's wrong? Issue #3740 is tracking invalid NDK paths that lead this this unhelpful error.

**Comment by stewbasic** [comment]
Yes, my issue was caused by not setting the Android ndk environment variable, so feel free to dupe this against #3740. However I was hoping the issue could be addressed more generally; ie could bazel at least say which rule caused the analysis phase to fail, even if the rule itself doesn't provide a useful error?

**Comment by jin** [comment]
This is fixed in ee33b63
